packages/react/src/drafts/SelectPanel2/SelectPanel.tsx

```diff
@@ -232,6 +241,10 @@ const Panel: React.FC<SelectPanelProps> = ({
         height="fit-content"
         maxHeight={maxHeight}
         sx={{
+          // to avoid a visible position shift, we delay the overlay animating-in
+          // to wait until the correct position is set (see useAnchoredPosition above for more)
+          animationDelay: '16ms',
```

Comment:
Note for reviewer: This isn't scientific at all, but we need to pick a number, so based it off a 60Hz refresh rate 😅 

From https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame, 

> The frequency of calls to the callback function will generally match the display refresh rate. The most common refresh rate is 60hz, (60 cycles/frames per second), though 75hz, 120hz, and 144hz are also widely used. requestAnimationFrame() calls are paused in most browsers when running in background tabs or hidden iframes, in order to improve performance and battery life.

Reply:
An alternative approach would be to "watch" for dialog open. Either in a recursive timeout or intersection observer, but that felt overkill!